src/Microsoft.Restier.Core/Query/QueryExpressionContext.cs

```diff
@@ -174,6 +174,9 @@ private QueryModelReference ComputeModelReference()
             QueryModelReference modelReference = null;
 
             var methodCall = this.VisitedNode as MethodCallExpression;
+            var parameter = this.VisitedNode as ParameterExpression;
+            var member = this.VisitedNode as MemberExpression;
```

Comment:
Thanks for the info. I prefer option 2. Better check each condition then return if fail.